src/pages/authPage/authUtils.js

```diff
@@ -0,0 +1,67 @@
+import axios from "axios";
+
+const signupHandler = async (signupDetails) => {
+  try {
+    const response = await axios.post(`/api/auth/signup`, signupDetails);
+    // saving the encodedToken in the localStorage
+    console.log(response.data.encodedToken);
+    localStorage.setItem("token", response.data.encodedToken);
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const signupReducer = (signupState, signupAction) => {
+  switch (signupAction.type) {
+    case "EMAIL_CHANGE":
+      return { ...signupState, email: signupAction.data.email };
+    case "PASSWORD_CHANGE":
+      return { ...signupState, password: signupAction.data.password };
+    case "CONFIRM_PASSWORD_CHANGE":
+      return {
+        ...signupState,
+        confirmPassword: signupAction.data.confirmPassword,
+      };
+    case "FIRST_NAME_CHANGE":
+      return { ...signupState, firstName: signupAction.data.firstName };
+    case "LAST_NAME_CHANGE":
+      return { ...signupState, lastName: signupAction.data.lastName };
```

Comment:
suggestion: you can keep a single action.type for all of them just change the payload.

Reply:
Wanted to keep logic in separate JS file